src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -326,20 +336,55 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
     }
 
     def annotations(): List[Tree] = {
-      //var annots = new ListBuffer[Tree]
+      var annots = new ListBuffer[Tree]
       while (in.token == AT) {
         in.nextToken()
-        annotation()
+        annots += annotation()
       }
-      List() // don't pass on annotations for now
+      annots.toList
     }
 
     /** Annotation ::= TypeName [`(` AnnotationArgument {`,` AnnotationArgument} `)`]
      */
-    def annotation() {
-      qualId()
-      if (in.token == LPAREN) { skipAhead(); accept(RPAREN) }
-      else if (in.token == LBRACE) { skipAhead(); accept(RBRACE) }
+    def annotation(): Tree = {
+      def annArg(): Tree = {
+        def annVal(): Tree = {
```

Comment:
Generally, we avoid any warnings for Java sources that run through the compiler.

Perhaps we should just plough ahead and add support for parsing `ConditionalExpression`. That's a necessary step to the full solution, but should be useful in and of itself, even before we can assign the right `ConstantType`.